lib/compress/zstd_compress.c

```diff
@@ -6327,7 +6327,7 @@ ZSTD_copySequencesToSeqStoreNoBlockDelim(ZSTD_CCtx* cctx, ZSTD_sequencePosition*
             /* Move to the next sequence */
             endPosInSequence -= currSeq.litLength + currSeq.matchLength;
             startPosInSequence = 0;
-            idx++;
+            idx++; /* Next Sequence */
         } else {
```

Comment:
I don't think this check is right when the `else` branch is taken and we split the sequence (we set `finalMatchSplit = 1;` on line 6352. Since we don't increment `idx` but still store a sequence.

You could simplify this a bit by moving the `idx++` to the bottom of the loop and doing:

```
if (!finalMatchSplit)
    ++idx;
```

Reply:
Oh thanks, great catch!